parachain/pallets/erc20-app/src/lib.rs

```diff
@@ -121,10 +121,9 @@ impl<T: Trait> Module<T> {
 }
 
 impl<T: Trait> Application for Module<T> {
-	fn handle(payload: &[u8], _verification_output: &VerificationOutput) -> DispatchResult {
-		let payload_decoded = InPayload::decode(payload)
+	fn handle(payload: &[u8], verification_output: &VerificationOutput) -> DispatchResult {
+		let payload_decoded = InPayload::decode_verified(payload, verification_output)
 			.map_err(|_| Error::<T>::InvalidPayload)?;
```

Comment:
not too familiar with Rust, but looks like even if there is an error here, it will still continue to process the payload, rather than fail? is that correct?

Reply:
Nope, the `?` operator basically means that the function will exit early with the `InvalidPayload` error as the result: https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html